### ISSUE TITLE
Exported column named types or exported column constants 

### ISSUE THREAD

**Issue by FourSigma** [vattle/sqlboiler#141]
#It would be nice if sqlboiler used schema specific named types for columns other than the golang **string** type. 

```
type Account struct {
	ProfileID    string `boil:"profile_id" json:"profile_id" toml:"profile_id" yaml:"profile_id"`
	Username     string `boil:"username" json:"username" toml:"username" yaml:"username"`
	Password     string `boil:"password" json:"password" toml:"password" yaml:"password"`
	Salt string `boil:"salt" json:"salt" toml:"salt" yaml:"salt"`
}

type accountColumn string

func(a accountColumn) String() string{
	return string(a)
}

const (
	ColumnAccountProfileId accountColumn  = "profile_id"
	ColumnAccountUsername  accountColumn  = "username"
	ColumnAccountPassword  accountColumn  = "password"
	ColumnAccountSalt      accountColumn  = "salt"
)

var (
	accountColumns               = []accountColumn{ColumnAccountProfileId, ColumnAccountUsername, ColumnAccountPassword, ColumnAccountSalt}
	accountColumnsWithoutDefault = []accountColumn{ColumnAccountProfileId, ColumnAccountUsername, ColumnAccountPassword, ColumnAccountSalt}
	accountColumnsWithDefault    = []accountColumn{}
	accountPrimaryKeyColumns     = []accountColumn{ColumnAccountProfileId}
)

```

Having typed columns significantly increase the type safety of database calls such as Insert and Update and   team members won't have to use error prone strings to make these calls. 

`err := account.Insert(db, models.ColumnAccountUsername, models.ColumnAccountPassword )`

`err := account.Update(db, models.ColumnAccountUsername, models.ColumnAccountPassword )`

This shouldn't too difficult of a refactor.  If you like the idea, I can start working on a pull request.

**Comment by aarondl** [comment]
This feature is accomplished by using MySQL and Postgres enums which we support: https://github.com/vattle/sqlboiler#enums

**Comment by aarondl** [comment]
Or is your idea that the enums should be typed as something else?

**Comment by FourSigma** [comment]
This feature would be unrelated to the Postgres/MySQL enums.  This is more from the standpoint of a bit more type safety.   For example: 

`
//Current Generated Code
func (o *Account) Update(exec boil.Executor, whitelist ...string) error
....
func (o *Account) UpsertG(updateOnConflict bool, conflictColumns []string, updateColumns []string, whitelist ...string) error
`

`
//Proposed Generated Code
func (o *Account) Update(exec boil.Executor, whitelist ...accountColumn) error
...
func (o *Account) UpsertG(updateOnConflict bool, conflictColumns []accountColumn, updateColumns []accountColumn, whitelist ...accountColumn) error
`

Basically type **string** and **[]string** would be replaced by named type **accountColumn**  and **[]accountColumn**.

**Comment by aarondl** [comment]
I fully understand this now. And the idea is actually pretty good.

There's two things that are not great about this:
- Currently not everywhere in the system could use this. For example a `Where()` clause couldn't use it because it's not generated code. This is something that I've thought about changing in a v4 maybe after looking at another Go ORM.
- The names get fairly long

This is also a breaking change so would have to be considered either optional or part of v3. Let's talk more about this one.

**Comment by aarondl** [comment]
@FourSigma This is done now, though it doesn't use a new type, it allows a small measure more safety than using plain strings. See the latest release https://github.com/vattle/sqlboiler/releases/tag/v2.5.0
